### ISSUE TITLE
`indent` rule is warning wrong on callbacks (master branch)

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#4142]
I'm using ESLint of `code-path-analysis/new` branch in my job. (to test the analysis).
Then I found indent rule's wrong warnings.
This is not reproduced in v1.6.0, but reproduced in master branch.

```js
/* eslint-env es6 */
/* eslint indent:[2,4] */
let foo = somethingList
    .filter(x => {
        return x;
    })
    .map(x => {
        return 100 * x;
    });
```

```
> git log -1
commit c1cd2b88c954e1ce24960126a50367885029f1db
...
> node .\bin\eslint.js test.js --no-ignore --no-eslintrc

C:\Users\t-nagashima.AD\Documents\GitHub\eslint\test.js
  5:9  error  Expected indentation of 4 space characters but found 8  indent
  6:5  error  Expected indentation of 0 space characters but found 4  indent
  8:9  error  Expected indentation of 4 space characters but found 8  indent
  9:5  error  Expected indentation of 0 space characters but found 4  indent

✖ 4 problems (4 errors, 0 warnings)
```

I expected no errors.
